pkg/virt-controller/watch/vmi.go

```diff
@@ -341,9 +344,41 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			log.Log.V(3).Object(vmi).Infof("All pods have been deleted, removing finalizer")
 			controller.RemoveFinalizer(vmiCopy, virtv1.VirtualMachineInstanceFinalizer)
 		}
-	case vmi.IsRunning() || vmi.IsScheduled():
-		// Don't process states where the vmi is clearly owned by virt-handler
+	case vmi.IsRunning():
+		// Keep PodReady condition in sync with the VMI
+		if cond := conditionManager.GetPodCondition(pod, k8sv1.PodReady); cond != nil {
+			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
+			conditionManager.AddPodCondition(vmiCopy, cond)
+		} else if conditionManager.HasCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady)) {
+			// Remove PodScheduling condition from the VM
+			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
+		}
+
+		// We don't own the object anymore, so patch instead of update
+		if !reflect.DeepEqual(vmiCopy.Status.Conditions, vmi.Status.Conditions) {
+			newConditions, err := json.Marshal(vmiCopy.Status.Conditions)
+			if err != nil {
+				return err
+			}
+			oldConditions, err := json.Marshal(vmi.Status.Conditions)
+			if err != nil {
+				return err
+			}
+			test := fmt.Sprintf(`{ "op": "test", "path": "/status/conditions", "value": %s }`, string(oldConditions))
+			patch := fmt.Sprintf(`{ "op": "replace", "path": "/status/conditions", "value": %s }`, string(newConditions))
+			_, err = c.clientset.VirtualMachineInstance(vmi.Namespace).Patch(vmi.Name, types.JSONPatchType, []byte(fmt.Sprintf("[ %s, %s ]", test, patch)))
+			// very ugly workaround. See https://github.com/kubernetes/kubernetes/issues/68202 for details
+			// If we have a test error, don't return an error, we will get an update with the latest version of the object
```

Comment:
Are we really saving much with this 'test' operation. Why not just return the patch error and try again next iteration until it eventually works?

I'd feel better about this if we weren't doing a string comparison with the error string to influence branching logic.

Reply:
Yep. Can remove the branch logic.